### PR TITLE
Api 966 r4 observation controller

### PR DESCRIPTION
**Pull request by jhulbert-va**
https://vajira.max.gov/browse/API-966

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java [hunk 1]
```diff
@@ -71,26 +71,44 @@ public Predicate toPredicate(
...
   @Value
-  class PatientAndCodesSpecification implements Specification<ObservationEntity> {
+  class PatientAndCodeAndDateSpecification implements Specification<ObservationEntity> {
     String patient;
 
     Set<String> codes;
 
+    DateTimeParameters date1;
+
+    DateTimeParameters date2;
```
Comment (taylorrassmannlibertyits) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java:81]: Does date1/date2 have more explanation behind them?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+ */
+@Validated
+@RestController
+@RequestMapping(
+    value = {"/r4/Observation"},
+    produces = {"application/json", "application/fhir+json"})
+public class R4ObservationController {
+
+  // ToDo /$validate
+  // ToDo SHOULD patient, category, and status
```
Comment (taylorrassmannlibertyits) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java:52]: Leftover TODOs?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java (hunk 1 above)
Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java:81]: Let's do some better names here

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java (hunk 2 above)
Comment (jhulbert-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java:52]: I purposefully left them there. Those endpoint will need to be added, but can't be right now. I will pull them out into a task though.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java (hunk 1 above)
Comment (jhulbert-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java:81]: That's just the convention we use for these specs. Searches that allow dates can have between 0 and 2 dates. I can't think of a better way to name those.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java (hunk 2 above)
Comment (jhulbert-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java]: https://vajira.max.gov/browse/API-1015

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java (hunk 1 above)
Comment (taylorrassmannlibertyits) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java:81]: I'm ok with keeping date1/date2 after further talks about it.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java (hunk 2 above)
Comment (taylorrassmannlibertyits) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java]: Now if only I could see the link to verify ;)

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java (hunk 1 above)
Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationRepository.java:81]: Fair enough, I forget that the dates parameters include comparison criteria (GT2005-01-21)

**Pull request merged**